### DISCUSSION TITLE
MySQL Unix sockets

### DISCUSSION THREAD

**Discussion by mps-pso** [enlightn/enlightn#193] [Q&A]
I'm not sure if this is a bug or I haven't implemented the change correctly and I was hoping for some help.

Enlightn keeps failing on check 11/67:

```
Check 11/67: MySQL is configured properly on single server setups. Failed
When MySQL is running on the same server as your app, it is recommended to use unix sockets instead of TCP ports to improve performance by upto 50% (Percona benchmark).
At config/database.php: line(s): 50.
```
I believe I have set up MySQL so it runs through a socket and my dbconfig is as follows:
```
{
    "host": "/var/run/mysqld/mysqld.sock",
    "socket": "/var/run/mysqld/mysqld.sock",
    "driver": "mysql"
}
```
My env variables:
```
DB_CONNECTION=mysql
DB_HOST=/var/run/mysqld/mysqld.sock
DB_PORT=null
DB_DATABASE=db_name
DB_USERNAME=forge
DB_PASSWORD="***"
DB_SOCKET=/var/run/mysqld/mysqld.sock
```
And my config:
```
'mysql' => [
            'driver' => 'mysql',
            'url' => env('DB_URL'),
            'host' => env('DB_HOST', '127.0.0.1'),
            'port' => env('DB_PORT'),
            'database' => env('DB_DATABASE', 'laravel'),
            'username' => env('DB_USERNAME', 'root'),
            'password' => env('DB_PASSWORD', ''),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset' => env('DB_CHARSET', 'utf8mb4'),
            'collation' => env('DB_COLLATION', 'utf8mb4_unicode_ci'),
            'prefix' => '',
            'prefix_indexes' => true,
            'strict' => true,
            'engine' => null,
            'options' => extension_loaded('pdo_mysql') ? array_filter([
                PDO::MYSQL_ATTR_SSL_CA => env('MYSQL_ATTR_SSL_CA'),
            ]) : [],
        ],
```
I'm pretty certain it's connecting via the socket but I'm at a loss. Can anyone help as to where I'm going wrong?

**Comment by ismaildasci** [comment]
Hey! 👋

It looks like you’re pretty close, but Enlightn might still be detecting a TCP setup due to a few subtle details. Here are a few things to try:

1. **Remove the `host` Parameter**: When using a Unix socket, sometimes the `host` field can cause confusion. Try setting `'host' => null` in `config/database.php` to make sure it's ignored.

2. **Set `DB_PORT` to an Empty String**: In your `.env` file, set `DB_PORT` to an empty string (`DB_PORT=""`). This clarifies that no TCP port is in use.

3. **Clear Config Cache**: After making changes, clear Laravel’s configuration cache with:
   ```bash
   php artisan config:clear
   ```

4. **Check the Actual Connection**: To double-check that Laravel is using the socket, you can add this line temporarily in a controller or command:
   ```php
   echo DB::connection()->getPdo()->getAttribute(PDO::ATTR_CONNECTION_STATUS);
   ```

This should help confirm if the connection is truly using the socket. Let me know if these steps solve it! 😊

**Reply by mps-pso** [reply]
> Hey! 👋
> 
> It looks like you’re pretty close, but Enlightn might still be detecting a TCP setup due to a few subtle details. Here are a few things to try:
> 
> 1. **Remove the `host` Parameter**: When using a Unix socket, sometimes the `host` field can cause confusion. Try setting `'host' => null` in `config/database.php` to make sure it's ignored.
> 2. **Set `DB_PORT` to an Empty String**: In your `.env` file, set `DB_PORT` to an empty string (`DB_PORT=""`). This clarifies that no TCP port is in use.
> 3. **Clear Config Cache**: After making changes, clear Laravel’s configuration cache with:
>    ```shell
>    php artisan config:clear
>    ```
> 4. **Check the Actual Connection**: To double-check that Laravel is using the socket, you can add this line temporarily in a controller or command:
>    ```
>    echo DB::connection()->getPdo()->getAttribute(PDO::ATTR_CONNECTION_STATUS);
>    ```
> 
> This should help confirm if the connection is truly using the socket. Let me know if these steps solve it! 😊

Hi @ismaildasci and thanks for taking the time to reply.

I have made your changes, albeit with a small change:

`'host' => null` generates an error when deploying:
```
In Connection.php line 825: Database hosts array is empty.

(Connection: mysql, SQL: select exists (select 1 from information_schema.tables where table_schema = 'my_schema' and table_name = 'migrations' and table_type in ('BASE TABLE', 'SYSTEM VERSI ONED')) as exists)

In ConnectionFactory.php line 210: Database hosts array is empty.
```
Changing it to an empty string: `'host' => ''` avoids that error.

It now works successfully, thank you for your help.
